uPortal-utils/uPortal-utils-core/src/main/java/org/apereo/portal/character/stream/events/PortletPlaceholderEventImpl.java

```diff
@@ -30,4 +32,37 @@ public PortletPlaceholderEventImpl(IPortletWindowId portletWindowId) {
     public final IPortletWindowId getPortletWindowId() {
         return this.portletWindowId;
     }
+
+    @Override
+    public int hashCode() {
+        int h = hash;
+        if (h == 0) {
+            h = internalHashCode();
+            hash = h;
+        }
+        return h;
+    }
+
+    protected int internalHashCode() {
+        final int prime = 31;
+        int result = 1;
+        result =
+                prime * result
+                        + ((this.getPortletWindowId() == null)
+                                ? 0
+                                : this.getPortletWindowId().hashCode());
+        result =
+                prime * result
+                        + ((this.getEventType() == null) ? 0 : this.getEventType().hashCode());
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return this.getClass().getSimpleName()
+                + " ["
+                + "portletWindowId="
+                + this.getPortletWindowId()
+                + "]";
```

Comment:
Would it make sense to leverage lombok for these?
https://projectlombok.org/features/EqualsAndHashCode
https://projectlombok.org/features/ToString

Reply:
Thanks @ChristianMurphy  for sharing this. Meanwhile, If my refactor change is fine,  May I request it to be merged ? 
I will explore this and update this part in a separate PR subsequently.
ref PR: https://github.com/uPortal-Project/uPortal/pull/2738